tools/c7n_oci/tests/test_identity.py

```diff
@@ -60,16 +60,40 @@ def test_identity_compartment(self, identity_compartment, test):
                     "op": "eq",
                 },
             ],
-            "actions": [
+            "actions": [{"type": "update", "freeform_tags": {"Environment": "Development"}}],
+        }
+        policy = test.load_policy(policy_str, session_factory=session_factory)
+        policy.run()
+        resource = self.fetch_validation_data(
+            policy.resource_manager, "get_compartment", new_compartment_id
+        )
+        assert resource is not None
+        test.assertEqual(resource["freeform_tags"]["Environment"], "Development")
+
+    @terraform("identity_compartment", scope="class")
+    def test_identity_update_compartment(self, identity_compartment, test):
+        compartment_id, new_compartment_id = self._get_identity_compartment_details(
+            identity_compartment
+        )
+        session_factory = test.oci_session_factory(
+            self.__class__.__name__, inspect.currentframe().f_code.co_name
```

Comment:
this pattern in the tests feels a bit strange, but it looks like it was already here. we could move the magic into the session factory to avoid the syntax for the caller.